game/songitems.cc

```diff
@@ -36,73 +42,54 @@ void SongItems::save(xmlpp::Element* songs) {
 
 SongId SongItems::addSongItem(std::string const& artist, std::string const& title, std::optional<SongId> _id) {
 	SongItem si;
-	si.id = _id.value_or(assign_id_internal());
```

Comment:
assign_id_internal() would always be evaluated here, as value_or can't short circuit